frontend/syntax.asdl

```diff
@@ -279,6 +279,12 @@ module syntax
     Word(word w)
   | YshExpr(expr e)
 
+  pat =
+    Else
+  | Words(List[word] w)
+  | YshExprs(List[expr] e)
```

Comment:
naming - I would use plural for the Lists

- w -> words
- e -> exprs

And then `r` could be `re eggex`